t/loc_tools.pl

```diff
@@ -245,8 +245,13 @@ (;$)
         my @local_categories_copy;
 
         if (ref $categories_ref) {
-            @local_categories_copy = @$$categories_ref;
-            $return_categories_numbers = 1;
+            if (ref $categories_ref eq 'ARRAY') {
+                @local_categories_copy = @$categories_ref;
+            }
+            else {
+                @local_categories_copy = @$$categories_ref;
```

Comment:
Not sure what this is supposed to do. If `$categories_ref` is not an arrayref then it will be a string.